docs/time_series_causal_graph.md

```diff
@@ -0,0 +1,350 @@
+# Time Series Causal Graph
+The class `cai_causal_graph.causal_graph.TimeSeriesCausalGraph`, inheriting directly from `cai_causal_graph.causal_graph.CausalGraph`,
+extends causal graphs to time series data.
+
+For the theoretical background of causality for time series, please refer to Chapter 10 of _Peters, J., Janzing, D. and Schölkopf, B., 2017. Elements of causal inference: foundations and learning algorithms (p. 288). The MIT Press_. 
+
+The main addition of `cai_causal_graph.causal_graph.TimeSeriesCausalGraph`
+lies in the type of nodes it supports.
+The new time series node is a new class `cai_causal_graph.causal_graph.TimeSeriesNode`
+inheriting from the corresponding `cai_causal_graph.causal_graph.Node` all the original attributes:
+- identifier
+- variable_type
+
+The node in a time series causal graph, `cai_causal_graph.causal_graph.TimeSeriesCausalGraph`,
+will have additional metadata that gives the time information of the node together with the variable name.
+The two additional metadata are:
+- `cai_causal_graph.type_definitions.TIME_LAG`: the time difference with respect to the reference time 0.
+- `cai_causal_graph.type_definitions.VARIABLE_NAME`: the name of the variable (without the lag information).
+
+Therefore, the time series node `cai_causal_graph.causal_graph.TimeSeriesCausalGraph` can be initialized as follows.
+
+```python
+from cai_causal_graph import TimeSeriesNode
+
+# Via identifier
+ts_node = TimeSeriesNode(identifier='X lag(n=1)')
+
+# Via time lag and variable name
+ts_node = TimeSeriesNode(time_lag=1, variable_name='X')
+
+# You can also provide the additional meta information
+ts_node = TimeSeriesNode(time_lag=1, variable_name='X', meta={TIME_LAG: 1, VARIABLE_NAME: 'X'})
+```
+
+## Examples of time series causal graphs
+The following examples have been drawn from _Peters, J., Janzing, D. and Schölkopf, B., 2017. Elements of causal inference: foundations and learning algorithms (p. 288). The MIT Press_.
+
+Example of a time series with no instantaneous effects.
+![ts_no_instantaneous_effects](images/ts_no_instantaneous_effects.png)
+
+Example of a time series with instantaneous effects.
+![ts_instantaneous_effects](images/ts_instantaneous_effects.png)
+
+Summary graph of the full time graphs above.
+![summary_graph](images/summary_graph.png)
+
+## Initialization
+
+### Direct initialization
+You can initialize the time series causal graph directly.
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+from cai_causal_graph.type_definitions import EdgeType
```

Comment:
Can be imported `from cai_causal_graph`. Put on same line as above. Fix throughout.